CustomGestures/CustomGestureRecognizer.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using CustomGestures.Core;
+using CustomGestures.Gestures;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Windows.Foundation;
+using Windows.UI.Input;
+
+namespace CustomGestures
+{
+    public class CustomGestureRecognizer
+    {
+        // Gesture events
+        public event TypedEventHandler<CustomGestureRecognizer, MultiTappedEventArgs> MultiTapped;
+
+        public CustomGestureSettings GestureSettings { get; set; }
+
+        private GestureProperties _gestureProperties;
+        private Dictionary<CustomGestureSettings, IGesture> _gestureProgress;
+        private Dictionary<uint, PointerInput> _inputs;
+
+        public CustomGestureRecognizer(GestureProperties gestureProperties = null)
+        {
+            _gestureProperties = gestureProperties;
+            _gestureProgress = new Dictionary<CustomGestureSettings, IGesture>();
+            _inputs = new Dictionary<uint, PointerInput>();
+        }
+
+        public void ProcessDownEvent(PointerPoint pointerPoint)
+        {
+            UpdatePointerEvent(pointerPoint, PointerEventType.Down);
+        }
+
+        public void ProcessMoveEvents(IList<PointerPoint> pointerPoints)
+        {
+            foreach (var pointerPoint in pointerPoints)
+            {
+                UpdatePointerEvent(pointerPoint, PointerEventType.Move);
+            }
+        }
+
+        public void ProcessUpEvent(PointerPoint pointerPoint)
+        {
+            UpdatePointerEvent(pointerPoint, PointerEventType.Up);
+
+            // Check if all touches are up. If so, clear the inputs.
+            bool allUp = true;
+            foreach (var input in _inputs.Values)
```

Comment:
Ooh, I like that! Good suggestion

Reply:
Yeah, it can even just go in the if statement, helps shorten the code as it's a simple thing you're doing; so you don't have to stare at it and wonder what the loop does ;)